tests/testthat/test-simulist-cleanepi.R

```diff
@@ -0,0 +1,67 @@
+set.seed(1)
+ll <- sim_linelist()
+
+test_that("convert_to_numeric corrects prop_int_as_word", {
+  messy_ll <- messy_linelist(
+    linelist = ll,
+    prop_missing = 0,
+    prop_spelling_mistakes = 0,
+    inconsistent_sex = FALSE,
+    numeric_as_char = FALSE,
+    date_as_char = FALSE,
+    prop_int_as_word = 0.5,
+    prop_duplicate_row = 0
+  )
+
+  clean_ll <- cleanepi::convert_to_numeric(
+    data = messy_ll,
+    target_columns = c("id", "age")
+  )
+
+  expect_false(identical(ll, clean_ll))
+  clean_ll[, c("id", "age")] <- apply(clean_ll[, c("id", "age")], MARGIN = 2, FUN = as.integer)
+
+  # check report is created as expected
+  report <- attr(clean_ll, "report")
+  expect_identical(names(report), "converted_into_numeric")
+  expect_identical(report$converted_into_numeric, "id, age")
+
+  # remove report to check identical line list <data.frame>
+  attr(clean_ll, "report") <- NULL
+
+  expect_identical(ll, clean_ll)
+})
+
+test_that("find_duplicates corrects prop_duplicate_row", {
+  messy_ll <- messy_linelist(
+    linelist = ll,
+    prop_missing = 0,
+    prop_spelling_mistakes = 0,
+    inconsistent_sex = FALSE,
+    numeric_as_char = FALSE,
+    date_as_char = FALSE,
+    prop_int_as_word = 0,
+    prop_duplicate_row = 0.2
+  )
+
+  clean_ll <- cleanepi::find_duplicates(data = messy_ll)
+
+  # check report is created as expected
+  report <- attr(clean_ll, "report")
+  expect_identical(names(report), c("duplicated_rows", "duplicates_checked_from"))
+
+  clean_ll <- cleanepi::remove_duplicates(data = messy_ll)
+
+  report <- attr(clean_ll, "report")
+  expect_identical(names(report), c("duplicated_rows", "duplicates_checked_from", "removed_duplicates"))
+
+
+  # remove appended column in cleaning
+  clean_ll$row_id <- NULL
```

Comment:
@Karim-Mane, is the addition of this column intended? Looking at the `cleanepi::remove_duplicates()` source code, it seems that the intent is to remove it before returning but this case is falling through the cracks.

Am I missing something or does this need tweaking in cleanepi?

Reply:
I raised this in epiverse-trace/cleanepi#225 and it was addressed in epiverse-trace/cleanepi#227, so I can update the test to remove this line and also depend on the dev version of {cleanepi} until the next release.

Reply:
Yes please, you can add cleanepi to the Remotes.

Reply:
Resolved in 9fb2b6d4cd6fe278abcde28c415099eb966cfa02.